packages/api-headless-cms/src/content/plugins/models.ts

```diff
@@ -50,18 +50,19 @@ export default () => {
         // Before continuing with the rest of the models, we must load the environment and assign it to the context.
         let environment = null;
         let environmentAlias = null;
-        if (context.cms.environment && typeof context.cms.environment === "string") {
-            if (context.models.CmsEnvironment.isId(context.cms.environment)) {
-                environment = await context.models.CmsEnvironment.findById(context.cms.environment);
-            } else {
-                environmentAlias = await context.models.CmsEnvironmentAlias.findOne({
-                    query: { slug: context.cms.environment }
-                });
-                environment = await environmentAlias.environment;
-            }
-        }
 
-        if (!environment) {
+        try {
```

Comment:
This won't work, because if the user passed an ID of unexisting env, line `57` would get executed, meaning `environment` would be `null` (`findById` will return `null`). After this, the next line we have is the `  context.cms.environment = environment.slug;`, which would again throw a mystery error to the user. :)

I'd stick with the proposition we discussed over the call:
![image](https://user-images.githubusercontent.com/5121148/86483049-7e78c280-bd53-11ea-8b0c-5031d517a8be.png)